helper/cert/source_gen.go

```diff
@@ -182,6 +199,33 @@ func (s *GenSource) checkLeader(ctx context.Context, changed chan<- bool) {
 	}
 }
 
+func (s *GenSource) retryUpdateSecret(ctx context.Context, bundle Bundle) {
+	defer func() {
+		s.StopUpdate <- struct{}{}
+	}()
+
+	// New exponential backoff with unlimited retries
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = time.Second * 30
+	bo.MaxElapsedTime = 0
+	ticker := backoff.NewTicker(bo)
+
+	for {
+		select {
```

Comment:
What happens if there is a leadership change during this retry loop? Should we also cancel then?

Reply:
Good question, and yes, I think we should also cancel then. Added in 51f676243f146b93c2836e9905f09ff69e15a00b.